src/next.md

```diff
@@ -7,6 +7,185 @@
+* Add relevant labels like `intermittent`, `regression`, `noreg-self` etc.
+* Set affects version to the JDK version(s) where the failure was seen.
+* In the description, always include (if possible):
+  * full name of the failing tests
+  * error messages
+  * assert messages
+  * stack trace
+  * command line information
+  * relevant information from the logs
+* Only set CPU and/or OS fields if the bug **ONLY** happens on that particular platform.
+* Always file separate bugs for different issues.
+  * If two crashes looks related but not similar enough to for sure be the same, it is easier to close a bug as a duplicate than it is to extract the relevant info from a bug to create a new one later.
+
+## ProblemListing or `@ignore`-ing a Test
+
+Sometimes tests break. It could be due to bugs in the test itself, or due to changed functionality in the code that the test is testing. While working on a fix, it can be useful to stop the test from being executed in everyone else's testing to reduce noise, especially if the test is expected to fail for more than a day. There are two ways to stop a test from being run in standard test runs: ProblemListing and using the `@ignore` keyword. Removing tests isn't the standard way to remove a failure. A failing test is expected to be a regression and should ideally be handled promptly with high urgency.
+
+I'll say it right away so that it's not forgotten at the end: Remember to remove the JBS id from the ProblemList or the test when the bug is closed. This is especially easy to forget if a bug is closed as a duplicate or 'Will Not Fix'.
+
+### ProblemListing jtreg tests
+
+ProblemListing is used for a short term exclusion while a test is being fixed, and for the exclusion of intermittently failing tests that cause too much noise, but can still be useful to run on an ad-hoc basis. ProblemListing is done in the file `ProblemList.txt`. There are actually several ProblemList files to choose from. Their location and name hint about what area or feature each file belongs to. Each file has sections for different components. All ProblemList files complement each other to build the total set of tests to exclude in JTReg runs.
+
+~~~
+test/hotspot/jtreg/ProblemList.txt
+test/hotspot/jtreg/ProblemList-aot.txt
+test/hotspot/jtreg/ProblemList-graal.txt
+test/hotspot/jtreg/ProblemList-non-cds-mode.txt
+test/hotspot/jtreg/ProblemList-Xcomp.txt
+test/hotspot/jtreg/ProblemList-zgc.txt
+test/jaxp/ProblemList.txt
+test/jdk/ProblemList.txt
+test/jdk/ProblemList-aot.txt
+test/jdk/ProblemList-graal.txt
+test/jdk/ProblemList-Xcomp.txt
+test/langtools/ProblemList.txt
+test/langtools/ProblemList-graal.txt
+test/lib-test/ProblemList.txt
+~~~
+
+Use the suitable ProblemList and add a line like this in the proper section:
+
+~~~
+foo/bar/MyTest.java                        4711   windows-all
+~~~
+
+In this example, `MyTest.java` is ProblemListed on windows, tracked by bug `JDK-4711`.
+
```

Comment:
Not sure where in this review to insert this but there is a sensible policy that you should make sure the bug report is open. If you are submitting it that is easy to ensure but there have been cases where bugs are closed (without necessarily being fixed) and the issue remains PL'd. A legitimate case where this occurs if a bug is fixed in one line of development but the fix is not (yet:?) backported and I'm not sure how the "sensible policy" accounts for that.

Reply:
This is a very good point. I'm not sure how to express it though since this is meant to be a document for everyone, not only Oracle developers.

I'm not sure what you mean with regards to the backport. If a fix is made in mainline it's removed from the ProblemList in mainline. The backport should remove it from the ProblemList in the update release once it's pushed there. The one thing that could be problematic is the `problemlist` label since that is only supposed to be on the main bug. But on the other hand I don't think there is any automation that looks at the ProblemLists in the update trains so having the label tied to the mainline issue won't be a problem in practice I think.

Reply:
I think Phil is talking about a separate line of development of current feature release, i.e. `jdk/jdk` vs `jdk/client`, and not about `jdk/jdk` vs `11u`. let's say you have fixed `JDK-4711` in `jdk/client` repo, but your fix hasn't been propagated to `jdk/jdk` yet, and folks in `jdk/jdk` are still affected by `JDK-4711` and see test failures (from non-problem-listed tests), so they want to go and problem-list them, but JBS says that `JDK-4711` is already closed.

Reply:
Ah, of course. There is no implemented solution for that right now. Two fairly easy options would be 1, Merge client into mainline, or 2, use a different fix version for client like done for project repos. This is something that we should bring up in a different thread with a different audience.